### PR TITLE
NEW : payment on several expense report (SQL integration)

### PR DESCRIPTION
**Pull request by atm-gauthier**

### CODE REVIEW

**Comment by atm-gauthier** [comment]
@eldy do you know why i get an error like this ?

image

File: htdocs/install/mysql/tables/llx_paymentuser.sql [hunk 1]
```diff
@@ -16,14 +17,14 @@
 --
 -- ===================================================================
 
-create table llx_payment_expensereport
+create table llx_paymentuser
```
[the remark this answers is not in the document]
Comment (atm-gauthier) [reply htdocs/install/mysql/tables/llx_paymentuser.sql:20]: No, i wanted to follow invoice system but just being able to pay several expensereport of same user.
On invoices, we have llx_paiement with total amount and llx_paiement_facture with link between payment and llx_facture (and in this case it allows to pay several invoices with same payment).

In this PR i did something like this : llx_paymentuser contains total amount and llx_payment_expense_report is the link between payment and llx_expensereport.

To help you, this is the new query of expensereport/payment/payment.php, now all approved expense report of $expensereport->fk_user_author are available for payment : 

	$sql = 'SELECT e.rowid, e.total_ttc, e.ref, SUM(pe.amount) as total_amount';
	$sql.= ' FROM '.MAIN_DB_PREFIX.'expensereport as e';
	$sql.= ' LEFT JOIN '.MAIN_DB_PREFIX.'payment_expense_report as pe ON pe.fk_expensereport = e.rowid';
	$sql.= ' WHERE fk_user_author = '.((int) $expensereport->fk_user_author);
	$sql .= ' AND e.entity IN ('.getEntity('expensereport').')';
	$sql .= ' AND e.fk_statut = '.ExpenseReport::STATUS_APPROVED;
	$sql .= ' AND e.paid = 0';
	$sql.= ' GROUP BY e.rowid, e.ref, e.total_ttc';
	$resql = $db->query($sql);

**Comment by atm-gauthier** [comment]
@eldy i get an error in ExpenseReportTest::testExpenseReportOther because my PR is splitted in 2
Code integration is here : https://github.com/Dolibarr/dolibarr/pull/23703

**Comment by atm-gauthier** [comment]
Hi @eldy, could you please give me a feedback about you are going to do with this PR ? Are you going to have a look at it later when you have time for it, or maybe it's not done as well as you would like to ?

**Comment by alexandre-janniaux** [comment]
https://github.com/Dolibarr/dolibarr/pull/23702#issuecomment-1405244365

> @eldy do you know why i get an error like this ?

It's what is fixed there, you can check the reason from phpunit there: #24621.

EDIT: missed the relevant line...

**Comment by atm-adrien** [comment]
@eldy

**Pull request closed without merging**